packages/lib/CalendarManager.ts

```diff
@@ -151,10 +151,70 @@ const cleanIntegrationKeys = (
   return rest;
 };
 
+/**
+ * This function deduplicates credentials based on selected calendars
+ * It removes regular credentials for which corresponding delegation credentials exist which can be identified only associating them with selected calendars
+ *
+ * This is important to prevent unnecessary/duplicate calls to calendar APIs
+ */
+export const deduplicateCredentialsBasedOnSelectedCalendars = ({
+  credentials,
+  selectedCalendars,
+}: {
+  credentials: CredentialForCalendarService[];
+  selectedCalendars: SelectedCalendar[];
+}) => {
+  // Only proceed if we have credentials
+  if (credentials.length === 0) {
+    return credentials;
+  }
+
+  // Get the user email from the first credential
+  const userEmail = credentials[0].user?.email;
+
+  // If no email, we can't identify which credential is duplicate
+  if (!userEmail) {
+    return credentials;
+  }
+
+  // Check if there are delegation credentials for the same integration types
+  const delegationCredentials = credentials.filter((credential) => credential.delegatedToId);
+
+  // Find all selected calendars with externalId matching the user's email and having a regular credential
+  const selectedCalendarsWithUserEmailConnectedWithRegularCredential = selectedCalendars.filter(
```

Comment:
We are only worried about these because delegation credential only has access to `userEmail`calendar